build_kernel.sh

```diff
@@ -10,62 +11,95 @@ OUTPUT_DIR=$DIR/output
 BOOT_IMG=./boot.img
 cd $DIR
 
+if [[ "$(uname)" == 'Darwin' ]]; then
+  BASE_VOLUME_PATH=$(echo $DIR | grep -o "^/Volumes/[^/]*" || echo "/")
+  if ! diskutil info -plist $BASE_VOLUME_PATH | grep -q "<string>Case-sensitive APFS</string>"; then
+    echo "---------------   macOS support   ---------------"
+    echo "Ensure you are in a Case-sensitive APFS volume to build the AGNOS kernel."
+    echo "https://github.com/commaai/agnos-builder?tab=readme-ov-file#macos"
+    echo "-------------------------------------------------"
+    exit 1
+  fi
+fi
+
+# Setup kernel build container
+echo "Building agnos-meta-builder docker image"
+docker build -f Dockerfile.builder -t agnos-meta-builder $DIR \
+  --build-arg UNAME=$(id -nu) \
+  --build-arg UID=$(id -u) \
+  --build-arg GID=$(id -g)
+echo "Starting agnos-meta-builder container"
+CONTAINER_ID=$(docker run -d -v $DIR:$DIR -w $DIR agnos-meta-builder)
```

Comment:
might fix the CI caching issue?
```suggestion
CONTAINER_ID=$(docker run -d -v $DIR:/tmp/builder -w /tmp/builder agnos-meta-builder)
```

Reply:
this was not the issue. the path is absolute in the GH action - always `/home/runner/work/agnos-builder/agnos-builder/`.

caching the "out" directory is completely ignored by the kernel build and I can't find how it's detecting that. nevertheless, I couldn't find any advice on caching the `out` directory. only ccache (or similar) approaches are recommended.

after I found the previous issue we had, ccache works much better now, will add a table with the build times in CI and on my machine.